tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -107,6 +107,13 @@ public void CloseSpan(Span span)
                 _spans.Add(span);
                 _openSpans--;
 
+                // If we close the local root span, make sure to unreference it in case we were to send another
+                // span in the same context later on (important in the case of AAS, to propagate resource.id)
+                if (span == RootSpan)
+                {
+                    RootSpan = default;
+                }
+
                 if (_openSpans == 0)
                 {
                     spansToWrite = _spans.GetArray();
```

Comment:
Maybe a simpler way to do this is to avoid modifying the `RootSpan` property and always call `DecorateWithAASMetadata(span);` inside the `if (spansToWrite.Count > 0)` to make sure the AAS spans are always added to any trace chunk. You could also rename `DecorateWithAASMetadata` to something like `AddMetadataToAllTraceChunks` or something like that?

Reply:
That's great thanks. I wish i had thought of that when I fixed the partial flush, I would have solved both bugs at once. Thank you